RevenueCatUI/Templates/V2/Components/Packages/PurchaseButton/PurchaseButtonComponentView.swift

```diff
@@ -83,6 +76,82 @@ struct PurchaseButtonComponentView: View {
         }
     }
 
+    private func purchase() async throws {
+        guard let action = self.viewModel.action else {
+            try await self.purchaseInApp()
+            return
+        }
+
+        switch action {
+        case .inAppCheckout:
+            try await self.purchaseInApp()
+        case .webCheckout:
+            try await self.purchaseWeb()
+        case .webProductSelection:
+            try await self.purchaseWebProductSelection()
+        }
+    }
+
+    private func purchaseInApp() async throws {
+        self.logIfInPreview(package: self.packageContext.package)
+
+        guard !self.purchaseHandler.actionInProgress else { return }
```

Comment:
If something is already running are we okay with the IAP purchase not going through?

Reply:
If this is the behavior we want, should we add this to the other purchase functions too?

Reply:
This is mainly when StoreKit is doing work so. So for web for now, it just opens to Safari so there isn't much to wait for 🤷‍♂️

Reply:
I think I've seen this in other components: `PuchaseButton` and `ButtonComponentView` do a similar thing, BUT only if you're going through `Purchases`.